src/App.tsx

```diff
@@ -184,8 +184,11 @@ const createRouter = () => {
         {
           path: createOpenSharedProjectPageUrl(),
           loader: ({ params }) => {
-            if (!params.shortId || !/^_[a-zA-Z0-9]+$/.test(params.shortId)) {
-              throw "Not a shared key";
+            if (
+              !params.shareId ||
+              !/^(_[a-zA-Z\d]{12})|(S(?:\d{5}-){3}\d{5})$/.test(params.shareId)
```

Comment:
This could be pulled out as a utility and have tests?

We don't have access to the backend code that produces these two type of ids to confirm that there is no additional variation we should cater for. Prefix of "_" is produced from users not logged in, prefix of "S" is from users who are logged in.